### PR TITLE
Migrate "Post.AnalyticsPostCount" to Sync by default

### PR DESCRIPTION
**Pull request by rvillablanca**
Fixes https://github.com/mattermost/mattermost-server/issues/10971

### CODE REVIEW

**Comment by rvillablanca** [comment]
I could not replicate the failed pipeline in jenkins. When I run `make check-style` I get the the next result

```shell
rodrigo@asus:/tmp/mattermost-server (fix-10971)$ make check-style 
Running GOVET
go get golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow
go: finding golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow latest
go: finding golang.org/x/tools/go/analysis/passes/shadow/cmd latest
go: finding golang.org/x/tools/go/analysis/passes/shadow latest
go: finding golang.org/x/tools/go/analysis/passes latest
go: finding golang.org/x/tools/go/analysis latest
go: finding golang.org/x/tools/go latest
go: finding golang.org/x/tools latest
go vet  github.com/mattermost/mattermost-server/api4 github.com/mattermost/mattermost-server/app github.com/mattermost/mattermost-server/cmd/mattermost github.com/mattermost/mattermost-server/cmd/mattermost/commands github.com/mattermost/mattermost-server/cmd/platform github.com/mattermost/mattermost-server/config github.com/mattermost/mattermost-server/einterfaces github.com/mattermost/mattermost-server/einterfaces/jobs github.com/mattermost/mattermost-server/imports github.com/mattermost/mattermost-server/jobs github.com/mattermost/mattermost-server/jobs/interfaces github.com/mattermost/mattermost-server/manualtesting github.com/mattermost/mattermost-server/migrations github.com/mattermost/mattermost-server/mlog github.com/mattermost/mattermost-server/mlog/human github.com/mattermost/mattermost-server/model github.com/mattermost/mattermost-server/model/gitlab github.com/mattermost/mattermost-server/plugin github.com/mattermost/mattermost-server/plugin/interface_generator github.com/mattermost/mattermost-server/plugin/plugintest github.com/mattermost/mattermost-server/plugin/plugintest/mock github.com/mattermost/mattermost-server/plugin/scheduler github.com/mattermost/mattermost-server/services/configservice github.com/mattermost/mattermost-server/services/filesstore github.com/mattermost/mattermost-server/services/filesstore/mocks github.com/mattermost/mattermost-server/services/httpservice github.com/mattermost/mattermost-server/services/imageproxy github.com/mattermost/mattermost-server/services/mailservice github.com/mattermost/mattermost-server/services/mfa github.com/mattermost/mattermost-server/services/timezones github.com/mattermost/mattermost-server/store github.com/mattermost/mattermost-server/store/sqlstore github.com/mattermost/mattermost-server/store/storetest github.com/mattermost/mattermost-server/store/storetest/mocks github.com/mattermost/mattermost-server/testlib github.com/mattermost/mattermost-server/utils github.com/mattermost/mattermost-server/utils/fileutils github.com/mattermost/mattermost-server/utils/imgutils github.com/mattermost/mattermost-server/utils/jsonutils github.com/mattermost/mattermost-server/utils/markdown github.com/mattermost/mattermost-server/utils/testutils github.com/mattermost/mattermost-server/web github.com/mattermost/mattermost-server/wsapi || exit 1
go vet -vettool=/home/rodrigo/go/bin/shadow  github.com/mattermost/mattermost-server/api4 github.com/mattermost/mattermost-server/app github.com/mattermost/mattermost-server/cmd/mattermost github.com/mattermost/mattermost-server/cmd/mattermost/commands github.com/mattermost/mattermost-server/cmd/platform github.com/mattermost/mattermost-server/config github.com/mattermost/mattermost-server/einterfaces github.com/mattermost/mattermost-server/einterfaces/jobs github.com/mattermost/mattermost-server/imports github.com/mattermost/mattermost-server/jobs github.com/mattermost/mattermost-server/jobs/interfaces github.com/mattermost/mattermost-server/manualtesting github.com/mattermost/mattermost-server/migrations github.com/mattermost/mattermost-server/mlog github.com/mattermost/mattermost-server/mlog/human github.com/mattermost/mattermost-server/model github.com/mattermost/mattermost-server/model/gitlab github.com/mattermost/mattermost-server/plugin github.com/mattermost/mattermost-server/plugin/interface_generator github.com/mattermost/mattermost-server/plugin/plugintest github.com/mattermost/mattermost-server/plugin/plugintest/mock github.com/mattermost/mattermost-server/plugin/scheduler github.com/mattermost/mattermost-server/services/configservice github.com/mattermost/mattermost-server/services/filesstore github.com/mattermost/mattermost-server/services/filesstore/mocks github.com/mattermost/mattermost-server/services/httpservice github.com/mattermost/mattermost-server/services/imageproxy github.com/mattermost/mattermost-server/services/mailservice github.com/mattermost/mattermost-server/services/mfa github.com/mattermost/mattermost-server/services/timezones github.com/mattermost/mattermost-server/store github.com/mattermost/mattermost-server/store/sqlstore github.com/mattermost/mattermost-server/store/storetest github.com/mattermost/mattermost-server/store/storetest/mocks github.com/mattermost/mattermost-server/testlib github.com/mattermost/mattermost-server/utils github.com/mattermost/mattermost-server/utils/fileutils github.com/mattermost/mattermost-server/utils/imgutils github.com/mattermost/mattermost-server/utils/jsonutils github.com/mattermost/mattermost-server/utils/markdown github.com/mattermost/mattermost-server/utils/testutils github.com/mattermost/mattermost-server/web github.com/mattermost/mattermost-server/wsapi || exit 1
go: warning: "./enterprise/..." matched no packages
Running GOFMT
Checking github.com/mattermost/mattermost-server/api4
Checking github.com/mattermost/mattermost-server/app
Checking github.com/mattermost/mattermost-server/cmd/mattermost
Checking github.com/mattermost/mattermost-server/cmd/mattermost/commands
Checking github.com/mattermost/mattermost-server/cmd/platform
Checking github.com/mattermost/mattermost-server/config
Checking github.com/mattermost/mattermost-server/einterfaces
Checking github.com/mattermost/mattermost-server/einterfaces/jobs
Checking github.com/mattermost/mattermost-server/imports
Checking github.com/mattermost/mattermost-server/jobs
Checking github.com/mattermost/mattermost-server/jobs/interfaces
Checking github.com/mattermost/mattermost-server/manualtesting
Checking github.com/mattermost/mattermost-server/migrations
Checking github.com/mattermost/mattermost-server/mlog
Checking github.com/mattermost/mattermost-server/mlog/human
Checking github.com/mattermost/mattermost-server/model
Checking github.com/mattermost/mattermost-server/model/gitlab
Checking github.com/mattermost/mattermost-server/plugin
Checking github.com/mattermost/mattermost-server/plugin/interface_generator
Checking github.com/mattermost/mattermost-server/plugin/plugintest
Checking github.com/mattermost/mattermost-server/plugin/plugintest/mock
Checking github.com/mattermost/mattermost-server/plugin/scheduler
Checking github.com/mattermost/mattermost-server/services/configservice
Checking github.com/mattermost/mattermost-server/services/filesstore
Checking github.com/mattermost/mattermost-server/services/filesstore/mocks
Checking github.com/mattermost/mattermost-server/services/httpservice
Checking github.com/mattermost/mattermost-server/services/imageproxy
Checking github.com/mattermost/mattermost-server/services/mailservice
Checking github.com/mattermost/mattermost-server/services/mfa
Checking github.com/mattermost/mattermost-server/services/timezones
Checking github.com/mattermost/mattermost-server/store
Checking github.com/mattermost/mattermost-server/store/sqlstore
Checking github.com/mattermost/mattermost-server/store/storetest
Checking github.com/mattermost/mattermost-server/store/storetest/mocks
Checking github.com/mattermost/mattermost-server/testlib
Checking github.com/mattermost/mattermost-server/utils
Checking github.com/mattermost/mattermost-server/utils/fileutils
Checking github.com/mattermost/mattermost-server/utils/imgutils
Checking github.com/mattermost/mattermost-server/utils/jsonutils
Checking github.com/mattermost/mattermost-server/utils/markdown
Checking github.com/mattermost/mattermost-server/utils/testutils
Checking github.com/mattermost/mattermost-server/web
Checking github.com/mattermost/mattermost-server/wsapi
gofmt success
go: warning: "./enterprise/..." matched no packages
./scripts/license-check.sh github.com/mattermost/mattermost-server/api4 github.com/mattermost/mattermost-server/app github.com/mattermost/mattermost-server/cmd/mattermost github.com/mattermost/mattermost-server/cmd/mattermost/commands github.com/mattermost/mattermost-server/cmd/platform github.com/mattermost/mattermost-server/config github.com/mattermost/mattermost-server/einterfaces github.com/mattermost/mattermost-server/einterfaces/jobs github.com/mattermost/mattermost-server/imports github.com/mattermost/mattermost-server/jobs github.com/mattermost/mattermost-server/jobs/interfaces github.com/mattermost/mattermost-server/manualtesting github.com/mattermost/mattermost-server/migrations github.com/mattermost/mattermost-server/mlog github.com/mattermost/mattermost-server/mlog/human github.com/mattermost/mattermost-server/model github.com/mattermost/mattermost-server/model/gitlab github.com/mattermost/mattermost-server/plugin github.com/mattermost/mattermost-server/plugin/interface_generator github.com/mattermost/mattermost-server/plugin/plugintest github.com/mattermost/mattermost-server/plugin/plugintest/mock github.com/mattermost/mattermost-server/plugin/scheduler github.com/mattermost/mattermost-server/services/configservice github.com/mattermost/mattermost-server/services/filesstore github.com/mattermost/mattermost-server/services/filesstore/mocks github.com/mattermost/mattermost-server/services/httpservice github.com/mattermost/mattermost-server/services/imageproxy github.com/mattermost/mattermost-server/services/mailservice github.com/mattermost/mattermost-server/services/mfa github.com/mattermost/mattermost-server/services/timezones github.com/mattermost/mattermost-server/store github.com/mattermost/mattermost-server/store/sqlstore github.com/mattermost/mattermost-server/store/storetest github.com/mattermost/mattermost-server/store/storetest/mocks github.com/mattermost/mattermost-server/testlib github.com/mattermost/mattermost-server/utils github.com/mattermost/mattermost-server/utils/fileutils github.com/mattermost/mattermost-server/utils/imgutils github.com/mattermost/mattermost-server/utils/jsonutils github.com/mattermost/mattermost-server/utils/markdown github.com/mattermost/mattermost-server/utils/testutils github.com/mattermost/mattermost-server/web github.com/mattermost/mattermost-server/wsapi
```
I'm trying to find what I could be doing wrong

**Comment by rvillablanca** [comment]
Could it be possible that jenkins is using some cache ?

**Comment by hanzei** [comment]
The build is failing because a change in the enterprise code is needed.
@lieut-data Would you be open on helping with this, given that Jesus is out?

**Comment by rvillablanca** [comment]
This is finished now, not a WIP

File: app/analytics.go [hunk 1]
```diff
@@ -58,9 +58,10 @@ func (a *App) GetAnalytics(name string, teamId string) (model.AnalyticsRows, *mo
 			})
 		}
 
-		var postChan store.StoreChannel
+		var count int64
+		var err *model.AppError
 		if !skipIntensiveQueries {
-			postChan = a.Srv.Store.Post().AnalyticsPostCount(teamId, false, false)
+			count, err = a.Srv.Store.Post().AnalyticsPostCount(teamId, false, false)
```
Comment (jespino) [inline app/analytics.go]: This is not the same, we would expect here a explicit async call (making the channel, running it inside a gorouting, etcetera. You can see it in the example PR).

**Comment by jespino** [comment]
@rvillablanca One thing, try to use descriptive names for the PRs just "Fix whatever" is hard to track when we look at the list of pending PRs. You can use the same name that is used in other PRs or in the Help Wanted ticket

File: app/analytics.go (hunk 1 above)
Comment (rvillablanca) [reply app/analytics.go]: I changed this and now I calls `AnalyticsPostCount` in the correct place

File: app/analytics.go [hunk 2]
```diff
@@ -205,31 +200,29 @@ func (a *App) GetAnalytics(name string, teamId string) (model.AnalyticsRows, *mo
 			close(sessionChan)
 		}()
 
-		var fileChan store.StoreChannel
-		var hashtagChan store.StoreChannel
+		var fileCount, hashtagCount int64
+		var fileError, hashtagError *model.AppError
 		if !skipIntensiveQueries {
-			fileChan = a.Srv.Store.Post().AnalyticsPostCount(teamId, true, false)
```
Comment (jespino) [inline app/analytics.go]: This is not the expected approach, the expected approach is to use the explicit async here, creating the channel and calling it in a gorouting (like you can see in the example PR). Not only for `fileChan`, for all the channels that you converted here into synchronous calls.

**Comment by rvillablanca** [comment]
I had to reorganize all in another branch because it was a little confusing still working on this PR.

I've created a new PR #11179

**Comment by jespino** [comment]
@rvillablanca We don't love this kind of changes because now I have to re-review the whole PR again, and actually, in this case I have to create a matching enterprise repo branch too.

**Comment by rvillablanca** [comment]
I am sorry, I won't do this again. So  what do you think about this? I can reopen this PR if you prefer or keep the other new PR.

**Comment by jespino** [comment]
No, the other PR is ok, I already reviewed that so... lets continue there :)

**Comment by rvillablanca** [comment]
Ok

**Pull request closed without merging**